### DISCUSSION TITLE
Duplocate synchronize events sent to github app when accepting " Suggested change"

### DISCUSSION THREAD

**Discussion by richtier** [community/community#24358] [Apps, API and Webhooks]
<p>When I commit a suggested comment e.g.,<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/2/1/2119e5e6fed39d7eb6af97fcf97191cde092fac5.png" data-download-href="/uploads/short-url/4IPf3wRxYAi2AdoYgufFeUUXxU9.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/5485798/181084345-5fb3de06-3442-4ee0-a2a2-c9931a96d9ab.png" alt="image" data-base62-sha1="4IPf3wRxYAi2AdoYgufFeUUXxU9" width="690" height="159" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/2/1/2119e5e6fed39d7eb6af97fcf97191cde092fac5_2_690x159.png, https://aws1.discourse-cdn.com/github/original/3X/2/1/2119e5e6fed39d7eb6af97fcf97191cde092fac5.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/2/1/2119e5e6fed39d7eb6af97fcf97191cde092fac5.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/2/1/2119e5e6fed39d7eb6af97fcf97191cde092fac5_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">792×183 15.1 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>then two synchronize events are sent to my github app within a second of each other</p>
<p>The body of each event is exactly the same.</p>
<p>My github app then performs the same action twice, which is not desirable.</p>
<p>Is this a bug with the webhooks? it seems like one</p>
<p>Is there a way to uniquely identify the events other than hashing the content into a “has this been seen befoore” key?</p>

**Comment by francisfuzz** [comment, marked as answer]
<p>:wave: <a class="mention" href="/richtier">@richtier</a>, thanks for reaching out and reporting this. I think it’s very interesting behavior to report and I don’t have an explanation for this. As a next step, I’m interested in reproducing this for myself. Here are my questions:</p>
<ul>
<li>Which permissions and webhook events are granted to your GitHub App?</li>
<li>How often do you see this behavior?</li>
<li>Are you able to reproduce this behavior in a public repository and share a URL here for us to investigate?</li>
</ul>
<blockquote>
<p>Is there a way to uniquely identify the events other than hashing the content into a “has this been seen befoore” key?</p>
</blockquote>
<p>When you receive a new webhook payload, you can inspect the headers and filter for the value of the <code>X-GitHub-Delivery</code>. This is a <a href="http://en.wikipedia.org/wiki/Globally_unique_identifier">GUID</a> to identify the delivery. Does that help with what you’re looking for, or are you looking to do something else?</p>

**Comment by richtier** [comment]
<p>sorry I thought I marked this as resolved. I was mistaken :slight_smile:</p>
